src/js/components/stock-movement-wizard/request/EditPage.jsx

```diff
@@ -1597,9 +1597,11 @@ class EditItemsPage extends Component {
               <button
                 type="button"
                 onClick={() => {
-                  window.location = !this.props.supportedActivities.includes('MANAGE_INVENTORY') && this.props.supportedActivities.includes('SUBMIT_REQUEST')
-                    ? stringUrlInterceptor('/dashboard')
-                    : stringUrlInterceptor('/stockMovement/list?direction=INBOUND');
+                  if (!this.props.supportedActivities.includes('MANAGE_INVENTORY') && this.props.supportedActivities.includes('SUBMIT_REQUEST')) {
+                    this.props.history.push(DASHBOARD_URL.base);
+                  } else {
+                    this.props.history.push({ pathname: STOCK_MOVEMENT_URL.list(), search: 'direction=INBOUND' });
```

Comment:
I get your explanation why you did it this way, but what I've figured out now is that we could maybe just add another function like `.listInbound()` which would look like this?
```javascript
 list: `${STOCK_MOVEMENT_URL.base}/list`,
 inboundList: () => `${STOCK_MOVEMENT_URL.list}/?direction=${StockMovementDirection.INBOUND}`,
 outboundList: () => `${STOCK_MOVEMENT_URL.list}/?direction=${StockMovementDirection.OUTBOUND}`,
 requestList: () => `${STOCK_MOVEMENT_URL.list}/?direction=${StockMovementDirection.OUTBOUND}&sourceType=${RequisitionSourceType.ELECTRONIC}`,
```
later it could be even improved by creating a method to invoke the search params into url instead of hardcoding them like I did, but for now I think it'd be fine.

What do you think guys about this version instead of doing it like this:
```js
this.props.history.push({ pathname: STOCK_MOVEMENT_URL.list(), search: 'direction=INBOUND' });
```
do it like this:
```js
this.props.history.push(STOCK_MOVEMENT_URL.inboundList());
```

Reply:
yes, cases like **inbound**, **outbound**, and **request** list makes sense since we treat them as separate pages.
I will implement it just like you showed in the example